gax/src/main/java/com/google/api/gax/batching/BatcherImpl.java

```diff
@@ -258,6 +268,7 @@ void onBatchSuccess(ResponseT response) {
 
     void onBatchFailure(Throwable throwable) {
       try {
+        batcherStats.recordBatchFailure(throwable);
```

Comment:
This should come after splitException

Reply:
Shifted `batcherStats.recordBatchFailure(throwable);` outside try/catch as we would be record failure in either of the case. I think `Batcher#recordBatchFailure` is just keeping a track of element so it is no needed to the inside try block.